### PR TITLE
UX: Provide entrypoint for survey

### PR DESCRIPTION
**Pull request by darkwing**
## **Description**

Adds the Survey toast to the home screen

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1650

## **Manual testing steps**

1. Open the home screen
2. See the survey
3. Click the close link
4. Refresh the extension
5. Don't see the survey again

### **Before**

N/A

### **After**
SCR-20231121-lkiy

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/home/home.container.js [hunk 1]
```diff
@@ -219,6 +222,10 @@ const mapDispatchToProps = (dispatch) => {
     setWaitForConfirmDeepLinkDialog: (wait) =>
       dispatch(mmiActions.setWaitForConfirmDeepLinkDialog(wait)),
     ///: END:ONLY_INCLUDE_IN
+    setSurveyLinkLastClickedOrClosed: (time) => {
+      console.log('dispatching setSurveyLinkLastClickedOrClosed with: ', time);
```
Comment (NidhiKJha) [inline ui/pages/home/home.container.js:226]: Nit: 
```suggestion
```

**Review by NidhiKJha — DISMISSED** [review]

File: ui/pages/home/home.component.js [hunk 2]
```diff
@@ -946,6 +954,28 @@ export default class Home extends PureComponent {
             }
           </div>
           {this.renderNotifications()}
+          {showSurveyToast ? (
+            <BannerBase
+              className="home__survey-banner"
+              backgroundColor={Color.iconDefault}
```
Comment (DDDDDanica) [inline ui/pages/home/home.component.js:960]:
```suggestion
              backgroundColor={BackgroundColor.iconDefault}
```

File: ui/pages/home/home.component.js [hunk 3]
```diff
@@ -946,6 +954,28 @@ export default class Home extends PureComponent {
...
+              titleProps={{ color: Color.backgroundDefault }}
```
Comment (DDDDDanica) [inline ui/pages/home/home.component.js:961]:
```suggestion
              titleProps={{ color: TextColor.backgroundDefault }}
```

File: ui/selectors/selectors.js [hunk 4]
```diff
@@ -1227,6 +1227,14 @@ export function getShowTermsOfUse(state) {
   );
 }
 
+export function getShowSurveyToast(state) {
+  const { surveyLinkLastClickedOrClosed } = state.metamask;
+  const startTime = new Date('December 20 2023 12:00:00 GMT-0600').getTime();
+  const endTime = new Date('December 20 2023 13:00:00 GMT-0600').getTime();
```
Comment (DDDDDanica) [inline ui/selectors/selectors.js:1233]: Maybe we can extract those 2 dates outside for easy and clear modification ?

File: ui/pages/home/home.component.js (hunk 2 above)
Comment (georgewrmarshall) [inline ui/pages/home/home.component.js:960]: You can update the entire component to use dark mode colors by using the data-theme attribute. I beleive it should be using dark theme colors no matter the theme
```suggestion
              data-theme="dark"
              backgroundColor={Color.iconDefault}
```

**Review by georgewrmarshall — COMMENTED** [review]
Looking good you can remove all the updates to the individual colors and use the data attr to update everything then it shouldn't matter what components are added

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Nice work. I pushed a small commit to lock the toast to dark mode using the `data-theme="dark"` attribute. This prevents us having to change every component color internally and now anything that appears in the toast will adhere to the dark mode colors. Attached is a recording showing the toast in different themes in popup and expanded view(code edited to show toast regardless of condition logic does not show behaviour well)

https://github.com/MetaMask/metamask-extension/assets/8112138/460db6ec-fecc-4354-9ec8-2757db5ffe8b

File: ui/pages/home/home.component.js (hunk 2 above)
Comment (georgewrmarshall) [reply ui/pages/home/home.component.js]: Updated here [ae964c9](https://github.com/MetaMask/metamask-extension/pull/21918/commits/ae964c94f250ba4f82df38ebb5861dd362c8a935)

File: ui/pages/home/home.component.js (hunk 3 above)
Comment (georgewrmarshall) [reply ui/pages/home/home.component.js]: Updated [ae964c9](https://github.com/MetaMask/metamask-extension/pull/21918/commits/ae964c94f250ba4f82df38ebb5861dd362c8a935)

File: ui/pages/home/home.component.js (hunk 2 above)
Comment (georgewrmarshall) [reply ui/pages/home/home.component.js]: Updated [ae964c9](https://github.com/MetaMask/metamask-extension/pull/21918/commits/ae964c94f250ba4f82df38ebb5861dd362c8a935)

**Review by vthomas13 — DISMISSED** [review]

File: app/scripts/controllers/app-state.js [hunk 5]
```diff
@@ -170,6 +171,12 @@ export default class AppStateController extends EventEmitter {
     });
   }
```
Comment (danjm) [inline app/scripts/controllers/app-state.js:173]: does the app-state controller persist state between sessions?

**Review by NidhiKJha — DISMISSED** [review]
LGTM

**Review by vthomas13 — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**